src/Stripes.test.js

```diff
@@ -0,0 +1,123 @@
+      });
+    });
+
+    describe('returns true', () => {
+      it('when the interface is present and no version is provided', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, discovery: { interfaces: { 'monkey': '3.0' }} });
+        expect(s.hasInterface('monkey')).toBe(true);
+        expect(logger.log).toHaveBeenCalledWith('interface', 'interface \'monkey\' exists');
+      });
+
+      it('when the requested version is present', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, discovery: { interfaces: { 'monkey': '3.0' } } });
+        expect(s.hasInterface('monkey', '3.0')).toBe('3.0');
+        expect(logger.log).toHaveBeenCalled();
+      });
+    });
+
+    it('returns false when the requested version is absent', () => {
+      const logger = { log: jest.fn() };
+      const s = new Stripes({ logger, discovery: { interfaces: { 'monkey': '3.0' }} });
+      expect(s.hasInterface('monkey', '4.0')).toBe(0);
+      expect(logger.log).toHaveBeenCalled();
+    });
+  });
+
+  describe('hasPerm', () => {
+    describe('returns true', () => {
+      it('given hasAllPerms', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({ logger, config: { hasAllPerms: true } });
+        expect(s.hasPerm('monkey')).toBe(true);
+        expect(logger.log).toHaveBeenCalledWith('perm', 'assuming perm \'monkey\': hasAllPerms is true');
+      });
+
+      it('when requested permissions are assigned', () => {
+        const logger = { log: jest.fn() };
+        const s = new Stripes({
+          logger,
+          user: {
+            perms: {
+              'monkey': true, 'bagel': true, 'funky': true, 'chicken': true
+            }
+          }
+        });
+        expect(s.hasPerm('monkey,bagel')).toBe(true);
+        expect(logger.log).toHaveBeenCalledWith('perm', 'checking perm \'monkey,bagel\': ', true);
+      });
+    });
+
+    it('returns false when requested permissions are not assigned', () => {
+      const logger = { log: jest.fn() };
+      const s = new Stripes({
+        logger,
+        user: {
+          perms: { 'monkey': true, 'bagel': true }
+        }
+      });
+      expect(s.hasPerm('monkey,funky')).toBe(false);
+      expect(logger.log).toHaveBeenCalledWith('perm', 'checking perm \'monkey,funky\': ', false);
+    });
+
+    it('returns undefined when user perms are uninitialized', () => {
+      const logger = { log: jest.fn() };
+      const s = new Stripes({ logger, user: {} });
+      expect(s.hasPerm('monkey')).toBeUndefined();
+      expect(logger.log).toHaveBeenCalledWith('perm', 'not checking perm \'monkey\': no user permissions yet');
+    });
+  });
+
+  describe('reset', () => {
+    it('restores config and logger to default values', () => {
+      const logger = { log: jest.fn() };
+      const s = new Stripes({ logger, config: { funky: 'chicken' } });
+      s.reset();
```

Comment:
I'm not sure I understand what the config is being reset _to_ in this case. Please advise.

Reply:
My intent was that `reset()` would restore default values, i.e. those pulled from `stripes-config` (for `config`) and from `./src/configureLogger.js` (for `logger`). 

As you pointed out above, however, my impl doesn't actually do that correctly. Imagining the impl were correct, however, how would you change line 113 to be more descriptive?

Reply:
I would be more inclined to set a value into the config, check that it's set, call `reset` and then check that the value is not set any longer.